server/routes/billRoute.js

```diff
@@ -78,11 +71,31 @@ router.post(
 	}
 );
 
-// @route   get app/bills
-// @desc    Get all the bill
+// @route   POST app/bill/update_dish
+// @desc    Update an existing dish
 // @access  Private
-router.get('/bills', auth, (req, res) => {
-	// res.json( {message: 'HI'});
-});
+router.post(
+	'/update_dish', 
+	[
+		auth,
+		check('billID', 'Bill ID is required').not().isEmpty(),
+		check('dishID', 'Dish ID is required').not().isEmpty(),
+		check('dishName', 'Dish name is required').not().isEmpty(),
+	],
+	async (req, res) => {
+		const billID = req.body.billID;
+		const dishID = req.body.dishID;
+		const dishName = req.body.dishName;
+
+		const targetDish = await Bill.findOne(
+			{_id: billID},
+			{dishes: {$elemMatch: {_id: dishID}}}
+		);
+		targetDish.dishes[0].dishName = dishName;
+		await targetDish.save();
+
+		res.json({message: 'Dish has been updated,'});
+	}
+)
```

Comment:
Added route to update a certain dish.

Reply:
could you wrap this in a try catch block similar to the ones I have in authRoute, make sure all the routes have one.

Reply:
Fixed.